eth_account/messages.py

```diff
@@ -14,14 +14,25 @@
 )
 
 
-def defunct_hash_message(primitive=None, hexstr=None, text=None):
+def defunct_hash_message(
+        *,
+        primitive=None,
+        hexstr=None,
+        text=None,
+        signature_version=b'E',
+        version_specific_data=None):
     '''
     Convert the provided message into a message hash, to be signed.
     This provides the same prefix and hashing approach as
-    :meth:`w3.eth.sign() <web3.eth.Eth.sign>`. That means that the
-    message will automatically be prepended with text
-    defined in EIP-191 as version 'E': ``b'\\x19Ethereum Signed Message:\\n'``
+    :meth:`w3.eth.sign() <web3.eth.Eth.sign>`. The message will by default
+    be prepended with text defined in EIP-191 as
+    version 'E': ``b'\\x19Ethereum Signed Message:\\n'``
     concatenated with the number of bytes in the message.
+    You can also specify the ``signature_version`` as
+        1) version ``0``: Sign data with intended validator (EIP 191)
+           Here the version_specific_data would be a hexstr which is the 20 bytes account address
+        2) version ``1``: Sign the structured data (EIP 712)
+           Here the version_specific_data would be <WIP(Needs to be updated)>
```

Comment:
The `version_specific_data` here is the domain separator https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#definition-of-domainseparator

If you are leaving out EIP 712 from this PR, I'd prefer to remove the WIP here, please :)